typing/typecore.ml

```diff
@@ -5522,10 +5525,10 @@ and map_half_typed_cases
         -> ty_infer:_ (* type to infer for body *)
         -> contains_gadt:_ (* whether the pattern contains a GADT *)
         -> ret)
-    -> partial_flag:bool
+    -> check_partiality:bool (* if false, assume Partial right away *)
```

Comment:
Would it make sense to rename the label to `check_if_total` or `assay_if_total`?

Otherwise one may assume that `check_partiality:false` means that the pattern was already deemed total and we don't need to check if the

Reply:
I didn't know about the word `assay`, so I'm not in favor. Okay for `check_if_total` -- I will rebase the PR.